js/stories.js

```diff
@@ -19,22 +19,52 @@ async function getAndShowStoriesOnStart() {
  * Returns the markup for the story.
  */
 
-function generateStoryMarkup(story) {
+function generateStoryMarkup(story, deleteBtn = false) {
   // console.debug("generateStoryMarkup", story);
 
   const hostName = story.getHostName();
   return $(`
       <li id="${story.storyId}">
-        <a href="${story.url}" target="a_blank" class="story-link">
-          ${story.title}
-        </a>
-        <small class="story-hostname">(${hostName})</small>
-        <small class="story-author">by ${story.author}</small>
-        <small class="story-user">posted by ${story.username}</small>
+        <div>
+          ${deleteBtn ? addDeleteBtn() : ""}
+          ${addStar(story, currentUser)} 
+          <a href="${story.url}" target="a_blank" class="story-link">
+            ${story.title}
+          </a>
+          <small class="story-hostname">(${hostName})</small>
+          <small class="story-author">by ${story.author}</small>
+          <small class="story-user">posted by ${story.username}</small>
+        </div>
       </li>
     `);
 }
 
+/** Adds star icon to story */
+
+function addStar(story, user) {
+  if (currentUser) {
+    const isFavorite = user.favorites.some((s) => s.storyId === story.storyId);
+    const starType = isFavorite ? "fas" : "far";
+    return `
+    <span class="star">
+      <i class="${starType} fa-star"></i>
+    </span>
+  `;
+  } else {
+    return "";
+  }
+}
```

Comment:
Since you're returning in the if statement already, you do not need the else, just return the empty string